analyses/npl/05.5_station_correlation_viz.md

```diff
@@ -0,0 +1,238 @@
+
+We want to understand what happens at Station B when Station A triggers according to a given threshold. 
+
+```python
+def make_max_rp_hist(ax, hist_vals, n_bins, title, xlab):
+    ax.hist(hist_vals, bins=n_bins)
+    ax.set_title(title)
+    ax.set_xlabel(xlab)
+    ax.axvline(1, c='red', ls='--')
+    ax.set_xlim([0, 2])
+    return ax
+
+def visualize_station_overlay(
+    da_primary, 
+    da_secondary, 
+    val_event, 
+    val_secondary, 
+    duration, 
+    plt_title, 
+    hist_title, 
+    save_title_line,
+    save_title_hist,
+    save=False):
+
+    groups = utils.get_groups_above_threshold(da_primary.values, val_event, duration)
+
+    if len(groups) == 0:
+        print('There are no events captured at this  threshold')
+        return
+    
+    max_post = []
+    max_during = []
+    max_pre = []
+
+    ngroups = len(groups)
+    nrows = 2
+    ncols = round(ngroups/nrows)
+    igroup = 1
+
+    fig, axs = plt.subplots(nrows, ncols, figsize=(15,5), squeeze=False, sharey=True)
+    fig.suptitle(plt_title)
+
+    for i in range(nrows):
+
+        for j in range(ncols):
+
+            try:
+                group = groups[igroup-1]
+                
+                # Values extend 15 days on either side of trigger event and normalize by the rp val
+                da_event_buffer = da_secondary[group[0]-15: (group[-1]+15)] / val_secondary
+                # Values just for the trigger event
+                da_event = da_secondary[group[0]: group[-1]+1]
+                # Values at the primary station for comparison
+                da_primary_buffer = da_primary[group[0]-15: (group[-1]+15)] / val_event
+                
+                # Keep track of the max value for each stage of the event
+                max_post.append(max(da_event_buffer[-15:-1].values))
+                max_during.append(max(da_event_buffer[15:-15].values))
+                max_pre.append(max(da_event_buffer[0:15].values))
+
+                ax = axs[i,j]
+                ax.axes.xaxis.set_ticks([])
+                ax.set_ylim([0, 1.5])
+
+                trigger_start_formatted = pd.to_datetime(da_event[0].time.values).date()
+                trigger_end_formatted = pd.to_datetime(da_event[-1].time.values).date()
+                ax.set_title(f'{trigger_start_formatted} - {trigger_end_formatted}', fontsize=8)
+                ax.fill_between(x=da_event.time, y1=0, y2=1.5,alpha=0.15)
+
+                ax.hlines(y=1, xmin=da_event_buffer.time[0], xmax=da_event_buffer.time[-1], lw=1, ls='--')
+                ax.plot(da_event_buffer.time, da_event_buffer.values, c='red', lw=1, alpha=1)
+                ax.plot(da_primary_buffer.time, da_primary_buffer.values, c='red',ls='--', lw=0.40, alpha=0.5)
+                ax.spines['right'].set_visible(False)
+                ax.spines['top'].set_visible(False)                
+
+            # In case we have too many subplots defined
+            except Exception as e:
```

Comment:
You should avoid catch-all exceptions if possible, here you can use `IndexError`